onnxruntime/core/providers/cpu/tensor/slice.cc

```diff
@@ -39,33 +76,33 @@ const T& clamp(const T& v, const T& lo, const T& hi) {
   return v;
 }
 }  // namespace
-Status SliceBase::PrepareForCompute(const size_t dimension_count, const std::vector<int64_t>& input_dimensions,
-                                    std::vector<int64_t>& starts, std::vector<int64_t>& output_dims) const {
+
+Status SliceBase::PrepareForCompute(const std::vector<int64_t>& raw_starts,
+                                    const std::vector<int64_t>& raw_ends, 
+                                    const std::vector<int64_t>& raw_axes,
+                                    const size_t                dimension_count,
+                                    const std::vector<int64_t>& input_dimensions,
+                                    std::vector<int64_t>&       starts,
+                                    std::vector<int64_t>&       output_dims) const {
   // Initialize axes to the provided axes attribute or to the default sequence
-  std::vector<int64_t> axes(axes_);
-  if (!has_axes_) {
+  std::vector<int64_t> axes(raw_axes);
+  if (axes.size() == 0) {
     //axes are omitted, they are set to[0, ..., ndim - 1]
     axes.resize(starts.size());
-    for (size_t i = 0; i < starts.size(); i++)
-      axes[i] = i;
-
-    if (axes.size() > starts_.size())
-      return Status(ONNXRUNTIME, INVALID_ARGUMENT, "'axes' has more entries than the 'starts' attribute holds");
-    if (axes.size() > ends_.size())
-      return Status(ONNXRUNTIME, INVALID_ARGUMENT, "'axes' has more entries than the 'ends' attribute holds");
+    std::iota(axes.begin(), axes.end(), 0);
   }
 
   // Iterate through the provided axes and override the start/end ranges
   for (size_t axesIndex = 0; axesIndex < axes.size(); axesIndex++) {
     auto axis = static_cast<size_t>(axes[axesIndex]);
```

Comment:
Please check sign before cast it to size_t.

Reply:
Thanks, but what cast?
If it is iota you are concerning about. this is its implementation:
template <class ForwardIterator, class T>
void iota (ForwardIterator first, ForwardIterator last, T val)
{
  while (first!=last) {
    *first = val;
    ++first;
    ++val;
  }//while
}

Reply:
auto axis = static_cast<size_t>(axes[axesIndex]);

Reply:
done, thanks!